### PR TITLE
Use API for mass SP migration

### PR DESCRIPTION
**Pull request by keichwa**
new section: SP mass migration

closes https://github.com/SUSE/spacewalk/issues/6684

### CODE REVIEW

File: modules/upgrade/pages/client-migration.adoc [hunk 1]
```diff
@@ -157,3 +157,36 @@ If the system status reports that the migration has failed, but also shows that
 After the migration is complete, schedule a reboot of the client from the {productname} Server.
 
 ////
+
+
+
+== SP Mass Migration
+
+If you want to migrate many clients to the next SP version, {suse} recommends to use {productname} API calls.
```
Comment (Loquacity) [inline modules/upgrade/pages/client-migration.adoc]:
```suggestion
If you want to migrate a large number of clients to the next SP version, you can use {productname} API calls.
```

File: modules/upgrade/pages/client-migration.adoc [hunk 2]
```diff
@@ -157,3 +157,36 @@ If the system status reports that the migration has failed, but also shows that
...
+
+You need to know to which migration target you want to migrate.
```
Comment (Loquacity) [inline modules/upgrade/pages/client-migration.adoc]: I think this should probably be a procedure.

File: modules/upgrade/pages/client-migration.adoc [hunk 3]
```diff
@@ -157,3 +157,36 @@ If the system status reports that the migration has failed, but also shows that
...
+List available migration targets:
+----
+spacecmd api -- system.listMigrationTargets -A 1000010001
+----
+Create a list of system IDs you want to migrate.
+
+Now we can iterate over the system IDs and call [systemitem]``listMigrationTarget`` for every system and check if the wanted target is part of the available targets.
```
Comment (Loquacity) [inline modules/upgrade/pages/client-migration.adoc]:
```suggestion
For each system ID, call [systemitem]``listMigrationTarget`` and check that it is an available target.
```

File: modules/upgrade/pages/client-migration.adoc [hunk 4]
```diff
@@ -157,3 +157,36 @@ If the system status reports that the migration has failed, but also shows that
...
+If yes, we call [systemitem]``system.scheduleSPMigration``, otherwise we skip the system.
```
Comment (Loquacity) [inline modules/upgrade/pages/client-migration.adoc]:
```suggestion
If the system ID is an available target, call [systemitem]``system.scheduleSPMigration``.
If it is not an available target, skip the system.
```

File: modules/upgrade/pages/client-migration.adoc [hunk 5]
```diff
@@ -157,3 +157,36 @@ If the system status reports that the migration has failed, but also shows that
...
+
+Here is a template that you can adapt for your environment:
```
Comment (Loquacity) [inline modules/upgrade/pages/client-migration.adoc]:
```suggestion
Adapt this template for your environment:
```

File: modules/upgrade/pages/client-migration.adoc (hunk 2 above)
Comment (keichwa) [reply modules/upgrade/pages/client-migration.adoc]: ok, trying to do my best ;)

**Pull request merged**